src/hooks/useNetworkStatus.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Custom hook that returns the browsers network status
+ */
+const useNetworkStatus = (): boolean => {
+    const [online, setOnline] = useState(navigator.onLine);
+
+    useEffect(() => {
+        window.addEventListener('online', () => setOnline(true));
+        window.addEventListener('offline', () => setOnline(false));
+
+        return () => {
```

Comment:
@Thenlie mind explaining this cleanup?

Reply:
If/when the hook un-mounts we remove the event listeners. This is just a standard pattern. I assume it very slightly improves performance on un-mount, but probably nothing noticeable.